classes/local/store/object_file_system.php

```diff
@@ -753,4 +753,28 @@ public function add_file_from_string($content) {
 
         return $result;
     }
+
+    /**
+     * Returns information about image.
+     * Information is determined from the file content.
+     * If file is external get image info by remote path.
+     *
+     * @param stored_file $file The file to inspect
+     * @return mixed array with width, height and mimetype; false if not an image
+     */
+    public function get_imageinfo(stored_file $file) {
+        if (!$this->is_image_from_storedfile($file)) {
+            return false;
+        }
+
+        $contenthash = $file->get_contenthash();
+        $location = $this->get_object_location_from_hash($contenthash);
+
+        if ($location === OBJECT_LOCATION_EXTERNAL) {
```

Comment:
I'm a bit worried about '===' here. If location will be returned as string '2' this if will return false.

Reply:
Actually scratch that. I looked at the code and get_object_location_from_hash will never return a string